core/services/workflows/workflow.go

```diff
@@ -11,37 +11,45 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/values"
 )
 
-type Capability struct {
+// yamlStep is the yaml parsed representation of a step in a workflow.
+type yamlStep struct {
 	Type   string         `yaml:"type"`
 	Ref    string         `yaml:"ref"`
 	Inputs map[string]any `yaml:"inputs"`
 	Config map[string]any `yaml:"config"`
 }
 
-type workflowSpec struct {
-	Triggers  []Capability `yaml:"triggers"`
-	Actions   []Capability `yaml:"actions"`
-	Consensus []Capability `yaml:"consensus"`
-	Targets   []Capability `yaml:"targets"`
+// yamlWorkflowSpec is the yaml parsed representation of a workflow.
+type yamlWorkflowSpec struct {
+	Triggers  []yamlStep `yaml:"triggers"`
+	Actions   []yamlStep `yaml:"actions"`
+	Consensus []yamlStep `yaml:"consensus"`
+	Targets   []yamlStep `yaml:"targets"`
 }
 
-func (w *workflowSpec) steps() []Capability {
-	s := []Capability{}
+func (w *yamlWorkflowSpec) steps() []yamlStep {
+	s := []yamlStep{}
 	s = append(s, w.Actions...)
 	s = append(s, w.Consensus...)
 	s = append(s, w.Targets...)
 	return s
 }
 
+// workflow is a directed graph of nodes, where each node is a step.
+//
+// triggers are special steps that are stored separately, they're
+// treated differently due to their nature of being the starting
+// point of a workflow. 
 type workflow struct {
-	graph.Graph[string, *node]
+	graph.Graph[string, *step]
 
 	triggers []*triggerCapability
 
-	spec *workflowSpec
+	spec *yamlWorkflowSpec
 }
 
-func (w *workflow) walkDo(start string, do func(n *node) error) error {
+// NOTE: should we make this concurrent?
```

Comment:
It's probably not worth it IMO since I'd expect the number of nodes to be fairly small (100s seems very high for number of steps). Also with the graph lib I don't think we have control of this.